kura/org.eclipse.kura.wire.script.filter.provider/src/org/eclipse/kura/wire/script/filter/provider/WiresScriptInterface.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *  Eurotech
+ *
+ *******************************************************************************/
+package org.eclipse.kura.wire.script.filter.provider;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.kura.type.TypedValue;
+import org.eclipse.kura.type.TypedValues;
+import org.eclipse.kura.wire.WireRecord;
+
+public class WiresScriptInterface {
+
+    private List<WireRecord> inputRecords;
+    private List<WireRecord> outputRecords;
+
+    public WiresScriptInterface(List<WireRecord> inputRecords) {
+        this.inputRecords = inputRecords;
+        this.outputRecords = null;
+    }
+
+    public OutputWireRecordInterface createRecord() {
+        return new OutputWireRecordInterface();
+    }
+
+    public void addRecord(OutputWireRecordInterface recordWrapper) {
+        if (outputRecords == null) {
+            outputRecords = new ArrayList<WireRecord>();
+        }
+        outputRecords.add(recordWrapper.getRecord());
+    }
+
+    public int getInputRecordCount() {
+        return inputRecords.size();
+    }
+
+    public WireRecordInterface getInputRecord(int i) {
+        if (i < 0 || i > inputRecords.size()) {
+            return null;
+        }
+        return new WireRecordInterface(inputRecords.get(i).getProperties());
+    }
+
+    public WireRecordInterface getFirstRecord() {
+        if (inputRecords.isEmpty()) {
+            return null;
+        }
+        return new WireRecordInterface(inputRecords.get(0).getProperties());
+    }
+
+    public static class WireRecordInterface {
+
+        Map<String, TypedValue<?>> properties;
+
+        private WireRecordInterface(Map<String, TypedValue<?>> properties) {
+            this.properties = properties;
+        }
+
+        public Object get(String key) {
+            TypedValue<?> value = this.properties.get(key);
+            if (value != null) {
+                return value.getValue();
+            }
+            return null;
+        }
+
+    }
+
+    public static class OutputWireRecordInterface extends WireRecordInterface {
+
+        public OutputWireRecordInterface() {
+            super(new HashMap<>());
+        }
+
+        public OutputWireRecordInterface putBoolean(String key, boolean value) {
```

Comment:
Instead of having methods for every types, you can use only one method, namely `put(String, boolean)` and use `TypedValues#newTypedValue(Object)`